### DISCUSSION TITLE
Connection errors in subscriptions and reconnects

### DISCUSSION THREAD

**Discussion by rhishikeshj** [obmarg/graphql-ws-client#89] [Q&A]
Hi,

I am using the `graphql-ws-client` with `async_tungstenite` and subscribing to multiple `subscriptions` to Hasura
I want to make these subscriptions resilient against network connection breaks etc.

My question is 2-fold

What is an effective way to introduce some websocket connection errors? 
I have tried the macOS network link conditioner with 100% loss but that doesn't seem to trigger any error behavior in `stream.next()`
I also tried turning the wifi completely off but that doesn't help either.

What if any would be good strategies to handle reconnects / re-subscribes? 
Do you have any ideas / opinions on this?

Thanks for your feedback.

**Comment by obmarg** [comment]
Sorry for the delay getting back to you on this @rhishikeshj - I've been swamped the past few weeks.

> What is an effective way to introduce some websocket connection errors?

I am not sure about this actually.  I'd have expected the network link conditioner to do the job, but perhaps graphql-ws-client is missing the dropped connections?  I do not have time to make the change, but if you wanted to make a PR that updates [the actor](https://github.com/obmarg/graphql-ws-client/blob/main/src/next/actor.rs) to periodically send [a ping](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#ping) that might fix it?

> What if any would be good strategies to handle reconnects / re-subscribes?
> Do you have any ideas / opinions on this?

Sorry, but I'm actually not sure on this one.  I can try and have a think, but as I said: I am very busy at the moment, and can't promise to get back to you.
